text/0931-template-compiler-api.md

```diff
@@ -289,6 +284,31 @@ When the `component` argument is passed, the return value is that backing class,
 > _Aren't route templates "bare templates"? What about them?<br>_
 > Yes, this RFC deliberately doesn't say anything about route templates. We expect a future routing RFC to use components to express what today's route templates express. This RFC also doesn't deprecate `precompileTemplate` yet -- although that will clearly be a good goal _after_ a new routing design addresses route templates.
 
+### Runtime Compilation
+
+Ember always had the ability to compile template at runtime. However, because this incur significant costs and most apps do not benefit from it, this feature is disabled by default and requires an explicit opt-in to include the runtime compiler.
+
+Traditionally, this is done with:
+
+```js
+// ember-cli-build.js
+app.import("node_modules/ember-source/dist/ember-template-compiler.js");
+```
+
+The new `"@ember/template-compiler/runtime"` module is interned to serve as a replacement for this, which better aligns with the direction we are headed. For example, this module can be imported on only the routes that needs it, and in conjunction with route-based code splitting that would reduce the performance hit on the initial load.
+
+Note that, even with the template compilation is available at runtime, the result of the compilations may be subtly different – applications may have custom glimmer/handlebars AST plugins in their build, and these plugins will not be available at runtime.
```

Comment:
it's probably not important to mention here, but for posterity, if someone wanted to add their template plugins they'd need to set up babel, similar to: https://github.com/NullVoxPopuli/limber/blob/main/packages/ember-repl/addon/src/browser/gjs.ts#L39

and pass their template plugins via: https://github.com/emberjs/babel-plugin-ember-template-compilation/ 's `transforms`